### DISCUSSION TITLE
Wallabag error "Communication with server failed" not sure what I'm doing wrong here

### DISCUSSION THREAD

**Discussion by davin900** [koreader/koreader#10847] [Q&A]
Using the latest KOReader (2023.06 I think) on a Kindle Paperwhite 5

So I finally got Wallabag up and running. I'm accessing it remotely via NGINX Proxy Manager and DuckDNS.

For some reason, KOReader on my PW5 seems to not be able to connect.

Here's what my settings look like on KOReader...

`-- we can read Lua syntax here!
return {
    ["wallabag"] = {
        ["articles_per_sync"] = 30,
        ["auto_tags"] = "",
        ["client_id"] = "copiedfromwallabag",
        ["client_secret"] = "alsocopiedfromwallabag",
        ["directory"] = "/mnt/us/books/wallabag/",
        ["download_queue"] = {},
        ["filter_tag"] = "",
        ["ignore_tags"] = "",
        ["is_archiving_deleted"] = true,
        ["is_auto_delete"] = true,
        ["is_delete_finished"] = true,
        ["is_delete_read"] = false,
        ["is_sync_remote_delete"] = true,
        ["password"] = "mypassword",
        ["remove_finished_from_history"] = false,
        ["send_review_as_tags"] = false,
        ["server_url"] = "https://wallabag.myurl.duckdns.org",
        ["username"] = "wallabag",
    },
}`

Obviously I've removed my actual password, client ID, secret, etc.

But the error I keep getting is this...

`172.17.0.1 - - [27/Aug/2023:01:20:23 +0000] "POST /oauth/v2/token HTTP/1.1" 400 94 "-" "KOReader/2023.06.1 (https://koreader.rocks/) LuaSocket/3.0.0"
`

I gather that the 400 error means I'm giving it incorrect info somehow... but I can't figure out what I'm doing wrong here. Should it be HTTPS? HTTP? I've tried both. Trailing slash after the URL? Enable websockets on NGINX? Any other ideas?

**Comment by Frenzie** [comment]
There are no WebSockets involved, but I would indeed be inclined to suspect the nginx config here.

**Reply by davin900** [reply]
So I decided to just bypass NGINX entirely to test this further... I changed my Wallabag URL to just a LAN IP (192.168.1.233:PORT), verified that URL works via browser, and put that in my URL field on KOReader.

I still get the same error. 

192.168.1.84 - - [27/Aug/2023:15:22:16 +0000] "POST /oauth/v2/token HTTP/1.1" 400 94 "-" "KOReader/2023.06.1 (https://koreader.rocks/) LuaSocket/3.0.0"

**Reply by NiLuJe** [reply]
Does the "we only support HTTP basic auth and not digest" thing applies here? (I mean, I'm seeing OAuth, so I think *not*, but this isn't really my wheelhouse).

**Reply by davin900** [reply]
You could be onto something. I finally got it to work using the LAN IP and creating a new client ID and client secret.

Now I think I’ll try the NGINX URL with HTTPS turned off.

**Reply by Frenzie** [reply]
@NiLuJe https works fine, the oauth stuff is how the Wallabag plugin / api works. The most likely mistake is a copy paste error, perhaps I should've lead with that but I got the impression that had been tried.

**Reply by davin900** [reply]
I swear I must've checked the copy/paste info a half dozen times but simply changing my password and creating a new client ID/secret fixed it.

Now it works great... even via NGINX! Is there a way to only make it sync the 30 most recent articles in my Wallabag list? It seems to do a random 30 at the moment.

thanks for your help

**Reply by Frenzie** [reply]
A… random 30? That doesn't meet how I remember things, but I might be misremembering or Wallabag itself may have changed, cf. <https://github.com/koreader/koreader/issues/10738#issuecomment-1657206603>.

**Reply by davin900** [reply]
Yeah I can't figure out what the 30 articles it pulled have in common... they certainly aren't all the most recent ones.
